mongorestore/mongorestore_test.go

```diff
@@ -583,3 +583,145 @@ func TestFixHashedIndexes(t *testing.T) {
 		})
 	})
 }
+
+func TestAutoIndexIdLocalDB(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	ctx := context.Background()
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	Convey("Test MongoRestore with {autoIndexId: false} in a local database's collection", t, func() {
+		dbName := session.Database("local")
+
+		// Drop the collection to clean up resources
+		defer dbName.Collection("test_auto_idx").Drop(ctx)
+
+		var args []string
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		restore.TargetDirectory = "testdata/local/test_auto_idx.bson"
+		result := restore.Restore()
+		So(result.Err, ShouldBeNil)
+
+		// Find the collection
+		filter := bson.D{{"name", "test_auto_idx"}}
+		cursor, err := session.Database("local").ListCollections(ctx, filter)
+		So(err, ShouldBeNil)
+
+		defer cursor.Close(ctx)
+
+		documentExists := cursor.Next(ctx)
+		So(documentExists, ShouldBeTrue)
+
+		var collInfo struct {
+			Options bson.M
+		}
+		err = cursor.Decode(&collInfo)
+		So(err, ShouldBeNil)
+
+		Convey("{autoIndexId: false} should not be flipped to true", func() {
+			if restore.serverVersion.GTE(db.Version{4, 0, 0}) {
```

Comment:
If we're only testing this for >= 4.0.0, I'd just skip the entire test at the start if it's >= 4.0.0.

Reply:
I actually don't know if we need to check the version here at all. If we're restoring to the `local` db, shouldn't `autoIndexId: false` always be preserved by restore, regardless of version?

Reply:
Ah, that's right Ryan. I'll remove the version check!